include/core/interface_tools.h

```diff
@@ -968,6 +969,9 @@ namespace InterfaceTools
     /// Solution vector of the signed distance (write only)
     LinearAlgebra::distributed::Vector<double> signed_distance;
 
+    /// Solution vector of the signed distance as used for output only
+    GlobalVectorType signed_distance_output;
```

Comment:
I'm not sure I understand, why do you need a GlobalVectorType for the output? Why not use directly the LinearAlgebra::distributed::Vector solutions below?
Is the limitation that the output must all be of the same vector type? If so, what happens when you do multiphysics with the matrix-free solver where all the physics are Trillinos vectors and the Navier-Stokes are regular vectors.

Or is the limitation that all of the output vector of a single physics must be of the same type?